### PR TITLE
Remove `SystemInfo()` error handling.

### PR DESCRIPTION
**Pull request by tao12345666333**
Signed-off-by: Jintao Zhang <zhangjintao9020@gmail.com>

**- What I did**

The `SystemInfo()` function never returns an error. 
Remove the `err` and clean up some error handing logics.

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by tao12345666333** [comment]
@thaJeztah  Can you help me run the job again ?  https://ci.docker.com/public/blue/organizations/jenkins/moby/detail/PR-39816/1/pipeline/39

**Comment by thaJeztah** [comment]
Looks like that node went offline before tests started;

```
01:53:03  azwin-2-32b280 was marked offline: Connection was broken: java.util.concurrent.TimeoutException: Ping started at 1567036143023 hasn't completed by 1567036383024
01:53:03  	at hudson.remoting.PingThread.ping(PingThread.java:134)
01:53:03  	at hudson.remoting.PingThread.run(PingThread.java:90)
01:53:03  
```

File: api/server/router/system/backend.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 // Backend is the methods that need to be implemented to provide
 // system specific functionality.
 type Backend interface {
-	SystemInfo() (*types.Info, error)
+	SystemInfo() *types.Info
```
Comment (thaJeztah) [inline api/server/router/system/backend.go]: Wondering if we should keep this a pointer or not

**Review by thaJeztah — COMMENTED** [review]
Thanks! Changes look good to me to get rid of the unused error

Given that this is changing an interface, I'm thinking if we can expect a change coming that would actually make us use the `error`. I did some refactoring a while back with in mind that I wanted to make this API more flexible (allow passing options to get only the information you're interested in, instead of always returning everything); https://github.com/moby/moby/blob/3042254a87274ff5e9561f2da1a986a703dfc60f/pkg/sysinfo/sysinfo_linux.go#L42-L48

https://github.com/moby/moby/blob/28678f2226b32cb61b417fb13c1b7245797064f3/daemon/info.go#L76-L82


I haven't had time to continue that work, but that would be my only concern (having to change the signature _again_ for those changes).


@cpuguy83 any thoughts?

File: api/server/router/system/backend.go (hunk 1 above)
Comment (tao12345666333) [reply api/server/router/system/backend.go]: Keep this a pointer +1. 
I tried to compare the difference between keep this a pointer or not with pprof, but the actual change is not big. So in theory, keep this a pointer may be more appropriate.

**Comment by tao12345666333** [comment]
> Given that this is changing an interface, I'm thinking if we can expect a change coming that would actually make us use the `error`. I did some refactoring a while back with in mind that I wanted to make this API more flexible (allow passing options to get only the information you're interested in, instead of always returning everything);

SGTM. But I may still need to know more detail.



> I haven't had time to continue that work, but that would be my only concern (having to change the signature _again_ for those changes).

If I can find out more information, I think I can help with this. :smile_cat:

**Comment by tao12345666333** [comment]
@thaJeztah Can you help me re-trigger CI job? Thanks!
And do I need to do anything else for this PR?

**Review by AkihiroSuda — APPROVED** [review]
LGTM

ping @thaJeztah

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Comment by AkihiroSuda** [comment]
Failure in z is unrelated. Merging.
```
[2020-03-12T03:46:00.101Z] --- FAIL: TestVolumesInspect (0.02s)

[2020-03-12T03:46:00.101Z]     volume_test.go:106: assertion failed: expression is false: createdAt.Truncate(time.Minute).Equal(now.Truncate(time.Minute)): CreatedAt (2020-03-12 03:45:59 +0000 UTC) not equal to creation time (2020-03-12 03:46:00.004349118 +0000 UTC m=+0.200139388)
```

**Pull request merged**